packages/synapse-client/package.json

```diff
@@ -0,0 +1,74 @@
+{
+  "name": "synapse-client",
+  "version": "0.0.1",
+  "description": "",
+  "type": "module",
+  "module": "./dist/index.js",
+  "types": "./dist/index.d.ts",
+  "exports": {
+    ".": {
+      "default": "./dist/index.js",
+      "types": "./dist/index.d.ts"
+    },
+    "./*": {
+      "default": "./dist/*.js",
+      "types": "./dist/*.d.ts"
+    }
+  },
+  "devDependencies": {
+    "@openapitools/openapi-generator-cli": "^2.13.5",
+    "@types/lodash-es": "^4.17.12",
+    "@types/minimist": "^1.2.5",
+    "@vitest/ui": "^1.6.0",
+    "lodash-es": "^4.17.21",
+    "minimist": "^1.2.8",
+    "rimraf": "^5.0.5",
+    "tslib": "^2.6.2",
+    "tsx": "^4.19.1",
+    "undici": "^6.19.4",
+    "vite": "^5.4.3",
+    "vite-config": "workspace:*",
+    "vitest": "^1.6.0"
+  },
+  "scripts": {
+    "format-spec": "prettier --write src/spec/openapispecification.json",
+    "get-spec:staging": "npx tsx src/codegen/downloadSpec.ts --stack staging && pnpm format-spec",
+    "get-spec:production": "npx tsx src/codegen/downloadSpec.ts --stack production && pnpm format-spec",
+    "get-spec": "pnpm get-spec:production",
+    "configure": "openapi-generator-cli version-manager set 7.8.0",
+    "generate": "MODEL_NAME_MAPPINGS=$(npx tsx src/codegen/generateModelNameMappings.ts); pnpm configure && rimraf src/generated && openapi-generator-cli generate -i src/spec/openapispecification.json -c config.json -o src/generated -t out/ --model-name-mappings $MODEL_NAME_MAPPINGS",
```

Comment:
Script to generate the TS code. First run the `generateModelNameMappings` script, which will scan the OpenAPI doc to modify the model names (e.g. `org.sagebionetworks.repo.model.FileEntity` -> `FileEntity`). Then run the openapi-generator, passing the generated mappings as an argument